packages/core/lib/console-child.js

```diff
@@ -1,11 +1,11 @@
-const Command = require("../lib/command");
 const TruffleError = require("@truffle/error");
 const Config = require("@truffle/config");
 const Web3 = require("web3");
 const yargs = require("yargs");
 
+// we split off the part Truffle cares about and need to convert to an array
 const input = process.argv[2].split(" -- ");
-const inputStrings = input[1];
+const inputStrings = input[1].split(" ");
```

Comment:
Why split on space? I don't know if it's possible in this code path, but if `inputStrings` were to have a value that's quoted spaces, this would break.

Reply:
The way it is structured now, `inputStrings` is an array with each member being the arguments passed to Truffle. In other words, if a user types `truffle compile --all false` this should be `["compile", "--all", "false"]`. This is what `getCommand` and `prepareOptions` takes. We could re-do how Truffle ingests this stuff if you have another idea, I haven't thought about it too much and tried to leave it similar to how it has been in the past.

Reply:
ok!